R/function_argument_linter.R

```diff
@@ -0,0 +1,29 @@
+#' Function argument linter
+#'
+#' Check that arguments with defaults come last in all function declarations.
+#'
+#' @evalRd rd_tags("function_argument_linter")
+#' @seealso
+#'   [linters] for a complete list of linters available in lintr.
+#' @export
+function_argument_linter <- function() {
+  xpath <- "//FUNCTION/following-sibling::EQ_FORMALS[1]/
+              following-sibling::SYMBOL_FORMALS[not(following-sibling::*[1][self::EQ_FORMALS])]"
```

Comment:
would a comment interfere with this `following-sibling::*[1]`?

Reply:
That would have to be a comment between the argument name and its equals sign. Is that even allowed? Never saw it in practice.

Reply:
I had this in mind:

```
function(
  really_long_argument_name = # TODO: deprecate this
    1
)
```

but = dropping to the next line should still parse

Reply:
You example would be safe. Bit it doesn't hurt to add the `not(self::COMMENT)` condition I think.

Note I don't have a lot of free time to work on lintr at the moment so it may take a while till I get to this.

Reply:
> Note I don't have a lot of free time to work on lintr at the moment so it may take a while till I get to this.

No problem at all. I also need to pull back from `lintr` relatively -- definitely over-committed in the push to get 3.0.0 out :)

Reply:
done.